src/main/java/in/zapr/druid/druidry/granularity/PeriodGranularity.java

```diff
@@ -39,5 +43,13 @@
 
     // TODO: Check if it is alright
     private DateTimeZone timeZone;
-    private Date origin;
+    private DateTime origin;
+
+    public String getOrigin() {
```

Comment:
Why is getter returning String?

Reply:
It has to. In JSON to druid we send String only. If we don't return String, the origin will return "origin":{"weekyear":2017,"yearOfEra":2017,"millisOfDay":68420218,"monthOfYear":11,"hourOfDay":19,"minuteOfHour":0,"secondOfMinute":20,"millisOfSecond":218,"weekOfWeekyear":46,"yearOfCentury":17,"centuryOfEra":20,"secondOfDay":68420,"minuteOfDay":1140,"year":2017,"dayOfMonth":13,"dayOfWeek":1,"era":1,"dayOfYear":317,"chronology":{"zone":{"fixed":false,"uncachedZone":{"fixed":false,"cachable":true,"id":"Asia/Kolkata"},"id":"Asia/Kolkata"}},"zone":{"fixed":false,"uncachedZone":{"fixed":false,"cachable":true,"id":"Asia/Kolkata"},"id":"Asia/Kolkata"},"millis":1510579820218,"afterNow":false,"beforeNow":true,"equalNow":false}